main/ejml-core/src/org/ejml/data/DMatrixSparseCSC.java

```diff
@@ -92,15 +89,17 @@ public DMatrixSparseCSC(int numRows , int numCols ) {
      * @param numCols Number of columns
      * @param arrayLength Initial maximum number of non-zero elements that can be in the matrix
      */
-    public DMatrixSparseCSC(int numRows , int numCols , int arrayLength) {
+    public DMatrixSparseCSC( int numRows, int numCols, int arrayLength ) {
+        if (UtilEjml.exceedsMaxMatrixSize(numRows, numCols))
+            throw new IllegalArgumentException("You have specified a matrix which exceeds the size of an integer");
```

Comment:
For sparse matrices this is actually valid to do so. The important point is, that the nz-values do not exceed Integer.MAX_VALUE